File: docs/node-mjs/playwright/link.py
from playwright.sync_api import sync_playwright

def get_all_links(url):
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()
        page.goto(url)
        # Extract hrefs via DOM evaluation
        hrefs = page.evaluate("""() => 
            Array.from(document.querySelectorAll('a'))
                .map(a => a.href)
                .filter(Boolean)
        """)
        browser.close()
        return hrefs

if __name__ == "__main__":
    links = get_all_links("https://example.com")
    print("Collected links:", links)

File: docs/node-mjs/playwright/link.mjs
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();       // or launch({ headless: false })
  const page = await browser.newPage();
  await page.goto('https://example.com');        // Replace with your target URL

  // Run DOM script to collect all hrefs
  const hrefs = await page.evaluate(() =>
    Array.from(document.querySelectorAll('a'))
      .map(a => a.href)
      .filter(Boolean)
  );

  console.log('Collected links:', hrefs);
  await browser.close();
})();
